torch_ort_inference/torch_ort/ortinferencemodule/ortinferencemodule.py

```diff
@@ -0,0 +1,239 @@
+        self._module_parameters = list(inspect.signature(self._original_module.forward).parameters.values())
+        self._device = _utils_infer.get_device_from_module(module)
+        self._export_mode = torch.onnx.TrainingMode.EVAL
+        self._export_extra_kwargs = {}
+        self._provider_options = provider_options
+        self._inference_session = None
+
+    def _forward_call(self, *inputs, **kwargs):
+        """Delegate the :meth:`~torch.nn.Module.forward` to
+        ONNX Runtime.
+
+        The first call to forward performs setup and checking steps. During this call,
+        ORTInferenceModule determines whether the module can be exported to ONNX. For
+        this reason, the first forward call execution takes longer than subsequent calls.
+        Execution is interrupted if ONNX Runtime cannot process the model for inferencing.
+
+        Args:
+            *inputs and **kwargs represent the positional, variable positional, keyword
+            and variable keyword arguments defined in the user's PyTorch module's forward
+            method. Values can be torch tensors and primitive types.
+
+        Returns:
+            The output as expected from the forward method defined by the user's
+            PyTorch module. Output values supported include tensors, nested sequences
+            of tensors and nested dictionaries of tensor values.
+        """
+        schema = _io._extract_schema({"args": copy.copy(inputs), "kwargs": copy.copy(kwargs)})
+        if (
+            self._onnx_models.exported_model
+            and schema == self._input_info.schema
+        ):
+            # All required models have already been exported previously
+            pass
+        else:
+            # Exporting module to ONNX for the first time
+            self._set_device_from_module(inputs, kwargs)
+            self._onnx_models.exported_model = self._export_model(schema,*inputs, **kwargs)
+            if self._debug_options.save_onnx_models.save:
+                self._onnx_models.save_exported_model(
+                    self._debug_options.save_onnx_models.path,
+                    self._debug_options.save_onnx_models.name_prefix,
+                    self._export_mode,
+                )
+
+            # Create the inference_session
+            if not self._inference_session:
+                session_options, providers, provider_options = self._get_session_config()
+                self._inference_session = onnxruntime.InferenceSession(
+                    self._onnx_models.exported_model.SerializeToString() , session_options, providers, provider_options
+                )
+
+        run_options = C.RunOptions()
+
+        # Pre-process inputs to make them compatible with onnxruntime
+
+        # Use IO binding
+        onnx_input_names = [inp.name for inp in self._onnx_models.exported_model.graph.input]
+        input_info = _io.parse_inputs_for_onnx_export(self._module_parameters, None, schema, inputs, kwargs)
```

Comment:
for inference scenarios, the input info should not change from one forward call to the next. This can be cached for performance. Again, this can be done in a follow up PR.